### PR TITLE
Drop electedness check in TransportClusterUpdateSettingsAction

### PR DESCRIPTION
**Pull request by DaveCTurner**
`TransportClusterUpdateSettingsAction` checks whether it's still the
elected master before submitting the followup reroute task. Such a check
can only be a best-effort thing because we might stand down as elected
master after the check, but before the task is executed. Morever it's
not necessary: if skipped, the reroute task will fail with a
`NotMasterException` yielding the same API response either way.

This commit removes the unnecessary electedness check.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java [hunk 1]
```diff
@@ -185,17 +185,6 @@ public void onAckTimeout() {
             }
 
             private void reroute(final boolean updateSettingsAcked) {
-                // We're about to send a second update task, so we need to check if we're still the elected master
```
Comment (kingherc) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java:188]: I'm not very experienced with the code here but if it's as simple as that the reroute task will return NotMasterException if the master changed in the meantime, it sounds good to me. I wonder @DaveCTurner if a test is possible (to ensure that it's handled correctly if the master changed in the meantime) or at least a comment to say that the following reroute task may fail with NotMasterException and it's fine.

**Review by kingherc — APPROVED** [review]
LGTM. One question. I am not expert with this code, so feel free to have a 2nd review if appropriate.

File: server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java:188]: (I think you might be right btw, it looks like the `BatchedRerouteService` in fact does not return a publish-failure exception on a publication failure)

**Review by tlrx — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java:188]: Ok I fixed that in #97224.

I'm not going to add tests in this PR. I have sketched out adding batching support to this action, which I'll open as a draft PR shortly to remind us to clean up this debt. If I added tests here too then they'd need to be reworked to handle the batching changes.

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/cluster/settings/TransportClusterUpdateSettingsAction.java:189]: See https://github.com/elastic/elasticsearch/pull/97246 for the batching PR

**Pull request merged**